app/boards/intel_adsp_ace15_mtpm.conf

```diff
@@ -79,6 +79,9 @@ CONFIG_INTEL_ADSP_IPC=y
 CONFIG_WATCHDOG=y
 CONFIG_LL_WATCHDOG=y
 
+# CONFIG_DMA_DW_SUSPEND_DRAIN=y needs reversed trigger order
+CONFIG_COMP_DAI_STOP_TRIGGER_ORDER_REVERSE=y
```

Comment:
Only ace15_mtpm selects CONFIG_DMA_DW_SUSPEND_DRAIN, it is only needed in that case.

Reply:
Humm, what differences existing on the DW/SSP side that would require a different behavior between generations? This looks rather weird to me.

Reply:
I really don't know, but the fact is that if the SSP is stopped, it will not trigger the DAM request so DW will not going to drain.
We don't see this issue on cAVS2.5 aqs the SUSPENSE_DRAIN is not enabled but we see it on MTL. LNL do not have it either, but it is not using DW anyways.

I cannot connect the dot why this change was added, what it suppose to fix and why it has not been blamed for this regression.
fwi, zephyr commit 6226f9e6e44f ("dma: dw: fix the return value check") made the drain timeout fatal, which imho should be a warn and we should be proceeding to disable the DMA channel force-ably

Reply:
Does the reverse order config depend on suspend_drain in the kconfig description?

Reply:
no

Reply:
> I really don't know, but the fact is that if the SSP is stopped, it will not trigger the DAM request so DW will not going to drain. We don't see this issue on cAVS2.5 aqs the SUSPENSE_DRAIN is not enabled but we see it on MTL. LNL do not have it either, but it is not using DW anyways.
> 
> I cannot connect the dot why this change was added, what it suppose to fix and why it has not been blamed for this regression. fwi, zephyr commit 6226f9e6e44f ("dma: dw: fix the return value check") made the drain timeout fatal, which imho should be a warn and we should be proceeding to disable the DMA channel force-ably

so this begs the question, why are we enabling CONFIG_DMA_DW_SUSPEND_DRAIN only for MTL? And what issue does enabling this manifest in our CI tests? Would enabling this on previous gens also result in the same issue?

Reply:
@ranj063, setting only `CONFIG_DMA_DW_SUSPEND_DRAIN` for TGL and I can reproduce the drain timeout.
The `CONFIG_COMP_DAI_STOP_TRIGGER_ORDER_REVERSE` does **not** fixes the issue.

This change in Zephyr can recover the system in case of drain timeout:
```diff
diff --git a/drivers/dma/dma_dw_common.c b/drivers/dma/dma_dw_common.c
index 22de9a7d9ef8..cd1076c686ed 100644
--- a/drivers/dma/dma_dw_common.c
+++ b/drivers/dma/dma_dw_common.c
@@ -594,10 +597,8 @@ int dw_dma_stop(const struct device *dev, uint32_t channel)
 	/* now we wait for FIFO to be empty */
 	bool fifo_empty = WAIT_FOR(dw_read(dev_cfg->base, DW_CFG_LOW(channel)) & DW_CFGL_FIFO_EMPTY,
 				DW_DMA_TIMEOUT, k_busy_wait(DW_DMA_TIMEOUT/10));
-	if (!fifo_empty) {
-		LOG_ERR("%s: dma %d channel drain time out", __func__, channel);
-		return -ETIMEDOUT;
-	}
+	if (!fifo_empty)
+		LOG_WRN("%s: dma %s channel %d drain time out", __func__, dev->name, channel);
 #endif
 
 	dw_write(dev_cfg->base, DW_DMA_CHAN_EN, DW_CHAN_MASK(channel));
@@ -606,7 +607,7 @@ int dw_dma_stop(const struct device *dev, uint32_t channel)
 	bool is_disabled = WAIT_FOR(!(dw_read(dev_cfg->base, DW_DMA_CHAN_EN) & DW_CHAN(channel)),
 				    DW_DMA_TIMEOUT, k_busy_wait(DW_DMA_TIMEOUT/10));
 	if (!is_disabled) {
-		LOG_ERR("%s: dma %d channel disable timeout", __func__, channel);
+		LOG_ERR("%s: dma %s channel %d disable timeout", __func__, dev->name, channel);
 		return -ETIMEDOUT;
 	}
 ```

The strange thing that I'm seeing on TGL:
hw:0,0 works
hw:0,1 is disabled in BIOS and it fails always the playback because the playback is not progressing.

Running this in one terminal:
```
test="aplay -q -d1 -Dhw:0,0 -fdat /dev/zero"
iteration=0

while :
do
        $test
        ((iteration++))
        echo "Iteration: ${iteration}"
        sleep 1
done
```

Then run this on another terminal:
```
aplay -q -d1 -Dhw:0,1 -fdat /dev/zero
```

We will have drain timeout **randomly**  on hw:0,0 or on hw:0,1 SSP.